draft-ietf-quic-tls.md

```diff
@@ -1116,9 +1125,22 @@ TLS KeyUpdate message.  Endpoints MUST treat the receipt of a TLS KeyUpdate
 message as a connection error of type 0x10a, equivalent to a fatal TLS alert of
 unexpected_message (see {{tls-errors}}).
 
-An endpoint MUST NOT initiate more than one key update at a time.  A new key
-cannot be used until the endpoint has received and successfully decrypted a
-packet with a matching KEY_PHASE.
+An endpoint MUST NOT initiate the first key update until the handshake is
+confirmed from its perspective. An endpoint MUST NOT initiate a subsequent
+key update until it has received an ACK for a packet sent at the previous
+KEY_PHASE.  This can be implemented as checking whether
+largest_recvd_ack_1rtt >= last_key_update_pn, where largest_recvd_ack_1rtt
+represents the highest value of the Largest Acknowledged field in any received
+1-RTT ACK frame, and where last_key_update_pn represents the lowest packet
+number that this endpoint sent using keys from the previous KEY_PHASE.
+
+Endpoints only need to maintain the two latest sets of packet protection keys
+and MAY discard older keys.  Updating keys multiple times rapidly can cause
+packets to be effectively lost if packets are significantly delayed.
+Therefore, an endpoint SHOULD NOT initiate a key update until three times the
+PTO after it has last updated keys. This avoids valid reordered packets being
+dropped by the peer because the peer may have discarded older keys, and reduces
+the computational overhead of key updates over the entire connection.
```

Comment:
I don't think that we need to talk about computational overhead here.

```suggestion
```

Reply:
Fair enough, fixed